Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,1111 @@
+    fHUDHeight[75] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[76] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[77] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[78] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[79] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[80] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[81] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[82] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[83] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[84] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[85] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[86] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[87] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[88] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[89] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[90] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[91] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[92] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[93] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[94] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[95] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[96] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[97] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[98] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[99] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[100] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[101] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[102] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[103] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[104] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+    fHUDHeight[105] = 0.002232143f * fWideScreenHeightScale * fHudHeightScale;
+
+    fHUDHeight[14] /= 1.125f;
+    fHUDHeight[15] /= 1.125f;
+    fHUDHeight[16] /= 1.125f;
+}
+
+void UpdateScreenAspectRatio()
+{
+    fWideScreenWidthScale = 640.0f / (*(pfScreenAspectRatio) * 448.0f);
+    fWideScreenHeightScale = 448.0 / 448.0f;
+    fWideScreenWidthProperScale = static_cast<float>(pRsGlobal->width) / (*pfScreenAspectRatio * 448.0f);
+    fWideScreenHeightProperScale = static_cast<float>(pRsGlobal->height) / 448.0f;
+    fDefaultWidth = (4.0f / 3.0f);
```

Comment:
This variable can be defined at the top as a constant

Reply:
I mean fDefaultWidth

Reply:
accepted